src/autoscaler/metricscollector/collector/app_streamer_test.go

```diff
@@ -174,20 +140,12 @@ var _ = Describe("AppStreamer", func() {
 					Timestamp:     fclock.Now().UnixNano(),
 				}))
 
-				go func() {
-					msgChan <- noaa.NewHttpStartStopEnvelope(333333, 100000000, 300000000, 1)
-					msgChan <- noaa.NewHttpStartStopEnvelope(555555, 300000000, 600000000, 1)
-					msgChan <- noaa.NewHttpStartStopEnvelope(666666, 300000000, 700000000, 1)
-				}()
-
-				By("collecting and computing throughput and responsetime for second interval")
-				Consistently(database.SaveMetricCallCount).Should(Equal(2))
-
-				By("save throughput and responsetime metric after the second collect interval")
-				fclock.Increment(TestCollectInterval)
-				Eventually(database.SaveMetricCallCount).Should(Equal(4))
+				msgChan <- noaa.NewHttpStartStopEnvelope(333333, 100000000, 300000000, 1)
+				msgChan <- noaa.NewHttpStartStopEnvelope(555555, 300000000, 600000000, 1)
+				msgChan <- noaa.NewHttpStartStopEnvelope(666666, 300000000, 700000000, 1)
+				fclock.WaitForWatcherAndIncrement(TestCollectInterval)
```

Comment:
check the channel is empty before you move the flock.. this will make sure the new metric is generated in the next interval